frontend/packages/console-shared/src/components/formik-fields/EnvironmentField.tsx

```diff
@@ -22,10 +26,10 @@ const EnvironmentField: React.FC<EnvironmentFieldProps> = ({
       metadata: { namespace },
     },
   } = props;
-  const { setFieldValue } = useFormikContext<FormikValues>();
+  const { setFieldValue, values } = useFormikContext<FormikValues>();
   const { t } = useTranslation();
   const fieldId = getFieldId(props.name, 'env-input');
-  const environmentVariables = !_.isEmpty(envs) ? envs.map((env) => _.values(env)) : [['', '']];
+  const environmentVariables = getEnvironmentVariables(envs);
```

Comment:
i think we memoize  it and use `environmentVariables` directly in `useEffect` as well

```suggestion
  const environmentVariables = React.useMemo(() => {
      return !_.isEmpty(envs) ? envs.map((env) => _.values(env)) : [['', '']];
  },[envs])
```

Reply:
alright, trying it now

regarding configMap key, can I make a new issue ? 
when we select the key on latest master branch also it doesn't appear

Reply:
yeah, we can log a new one for that.

Reply:
I believe the issue was caused because a prop had the name `key` which is reserved by React
changed the prop to `pairKey`